newlib/libc/tinystdio/vfprintf.c

```diff
@@ -1139,6 +1174,10 @@ int vfprintf (FILE * stream, const CHAR *fmt, va_list ap_orig)
 #endif
 #ifdef _PRINTF_PERCENT_N
```

Comment:
This needs to be `defined(_PRINTF_PERCENT_N) || defined(VFPRINTF_S)` so that we always get this error message when using VFPRINTF_S, and not just when the library is built to support `%n` in vfprintf.